Automattic/wp-calypso
# Jetpack Backup: Update component to use custom backup retention days if set

#### Proposed Changes

* Update the component to make use of a custom backup retention period (in days) if set by the customer otherwise use the retention from the plan.

Requires D100421-code on VP.

#### Testing Instructions

1) Take a site whose backups are being discarded due to the storage limit has been reached.
2) Check the current backup dashboard status, it should be similar to the below screenshot. Ensure that no custom retention has been set yet or delete if it is, from the vpshell.
     <img width="784" alt="Screenshot 2023-02-07 at 5 49 05 PM" src="https://user-images.githubusercontent.com/13975226/217243234-e1b72f81-385f-46cb-a4de-ddc77ba61977.png">
3) Now set a custom retention period lower than that of the current `days_of_backups_allowed` from the size API either by following D100335-code or via from vpshell (e.g. in D100421-code)
4) Refresh the Backup Dashboard on Calypso, it should show normal usage with no warning message as below:
     <img width="763" alt="Screenshot 2023-02-07 at 5 55 28 PM" src="https://user-images.githubusercontent.com/13975226/217244289-a4d1c15b-9fed-4f0b-9ed3-03592a5f6805.png">

Also, ensure unit tests passes by executing `yarn run test-client client/state/rewind`

#### Pre-merge Checklist





Related to D100421-code
